test/lit/passes/merge-blocks.wast

```diff
@@ -31,4 +38,56 @@
    )
   )
  )
+
+ ;; CHECK:      (func $struct.set
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (i32.const 1234)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (struct.set $struct 0
+ ;; CHECK-NEXT:   (ref.null $struct)
+ ;; CHECK-NEXT:   (i32.const 5)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT: )
+ (func $struct.set
+  (block
+   (nop)
```

Comment:
Why are these necessary? And does this pass not only push `block`s outside but also remove unnecessary `block`s as well?

Reply:
The nop is just to show more content in the outer block. Not necessary I guess, but I think it's a little clearer, as they remain afterwards, so we can see things in relation to them. But I can remove if you prefer?

Yes, the pass will merge blocks when it can. In fact the pushing of blocks outside is primarily in order to enable that merging, which is often possible.